src/main/java/com/marianbastiurea/lifeofbees/Queen.java

```diff
@@ -20,8 +20,34 @@ of her life, the new queen will kill her (called 'supercedure').  If the old que
  */
 
 public class Queen {
+    private int ageOfQueen;
 
-    public double makeEggs(Hive hive, Environment environment) {
-        return hive.getHoneyFrames() * environment.getWeather().getTemperature()* 0.15 ;
+    public int getAgeOfQueen() {
+        return ageOfQueen;
     }
+
+    public void makeEggs(Hive hive, Environment environment) {
+        int numberOfEggs = (int) (2000 */* honeyIndex * */environment.getWeather().getTemperature()* ageOfQueenIndex(ageOfQueen));
+
+        hive.addEggs(numberOfEggs) ;
+    }
+    public static double ageOfQueenIndex(int ageOfQueen) {
```

Comment:
Remove `static` and use `ageOfQueen` property  instead of `ageOfQueen` parameter.